python/cudf/cudf/core/column/string.py

```diff
@@ -1455,17 +1455,13 @@ def endswith(self, pat, **kwargs):
                 as cudf uses native strings instead of Python objects"
             )
 
-        from cudf._libxx.scalar import Scalar
-
-        # TODO: Cleanup if/else blocks after this issue is fixed:
-        # https://github.com/rapidsai/cudf/issues/4500
-        if pat == "":
-            result_col = column.as_column(
-                True, dtype="bool", length=len(self._column)
-            ).set_mask(self._column.mask)
-        elif pat is None:
-            result_col = column.as_column(np.nan, length=len(self._column))
+        if pat is None:
+            result_col = column.column_empty(
+                len(self._column), dtype="float64"
```

Comment:
Can we keep this as `bool` instead of `float64`?

Reply:
Keeping `bool` would actually return a column with all `True` values instead of `null`
```python
    def test_string_starts_ends(data, pat):
        ps = pd.Series(data)
        gs = Series(data)
    
>       assert_eq(ps.str.startswith(pat), gs.str.startswith(pat))
E   AssertionError: Series are different
E   
E   Series values are different (100.0 %)
E   [left]:  [nan, nan, nan, nan]
E   [right]: [True, True, True, True]
```

Reply:
You'd need to add `masked=True` to the `column_empty` call

Reply:
Thanks, changed it. 👍